### ISSUE TITLE
Allow user to create Tasks with the Windows 'Jump List' functionality

### ISSUE THREAD

**Issue by DanAtkinson** [Microsoft/vscode#34705]
No matter how many projects and workspaces I've had open, my Tasks entries are always blank.

![2017-09-20 16_57_48-clipboard](https://user-images.githubusercontent.com/23440/30654176-134bc0ac-9e25-11e7-8e30-e0eac6459687.png)

I've tried dragging folders to them but it doesn't work (I get a 'not allowed' cursor icon). So it seems as if this functionality is redundant.

Further, in Code itself, there appears to be no mechanism for saving the current workspace as a 'Task' in the taskbar for easy retrieval later. This would be especially useful for me as it could then be used to maintain a state for disparate files that aren't connected to a single workspace.

This functionality would probably also be useful for users who are used to the Notepad++ way of managing sessions - something which Code doesn't seem to support or allow (hotExit doesn't work with unsaved documents).

**Comment by DanAtkinson** [comment]
No, vscodebot, this is **not** a [`tasks`](https://github.com/Microsoft/vscode/labels/tasks) issue!

**Comment by bpasero** [comment]
That is weird, we do fill task entries:

![image](https://user-images.githubusercontent.com/900690/30679836-7c62893c-9e9c-11e7-92b4-283e1c99206b.png)

Did you properly install VS Code?

Can you try to reproduce with our nightly insider builds? You can give our preview releases a try from: http://code.visualstudio.com/Download#insiders

**Comment by DanAtkinson** [comment]
@bpasero Hi. I've definitely installed it correctly. I re-installed it and still have the same issue.

I also downloaded the insider edition, opened a folder, saved that as a workspace file (nice!), closed Code, re-opened (the workspace was still there). Opened other files outside this workspace. Still nothing.

As I'm writing this, this is what my taskbar shows for the Insiders app. Just New Window but no workspaces or existing files

![image](https://user-images.githubusercontent.com/23440/30703625-2673d138-9ee8-11e7-9cf2-0858f8e37a8f.png)

Information on my Code and Insiders for reference.
![image](https://user-images.githubusercontent.com/23440/30703792-96bc3c96-9ee8-11e7-9e7b-7dad9a2057c8.png)
![image](https://user-images.githubusercontent.com/23440/30703771-894a8f22-9ee8-11e7-86bd-a703f7dad927.png)

Thanks, Dan

**Comment by bpasero** [comment]
What Windows version is this? Some insiders update?

**Comment by DanAtkinson** [comment]
@bpasero No, It's Windows 10 Pro build v1511 (OS Build 10586.1106).

**Comment by bpasero** [comment]
Please see if you can find any logged output that could help us find the source of the issue:
* run `code --verbose` from the command line and attach the output
* in Code select **Help | Toggle Developer Tools** and select the **Console**
* attach any output you see in the **Console** you can find

Thanks!

**Comment by DanAtkinson** [comment]
Here is the output of `code --verbose > code-debug.txt` - [code-debug.txt](https://github.com/Microsoft/vscode/files/1324456/code-debug.txt).

Below is the console output. NB, this is from Code, not Code Insiders. I can also provide that if requested.

    keybindingsRegistry.ts:197 Ctrl+Alt+ keybindings should not be used by default under Windows. Offender:  e  for  xmlTools.formatXml
    e._assertNoCtrlAlt @ keybindingsRegistry.ts:197

    keybindingsRegistry.ts:197 Ctrl+Alt+ keybindings should not be used by default under Windows. Offender:  e  for  xmlTools.evaluateXPath
    e._assertNoCtrlAlt @ keybindingsRegistry.ts:197

    keybindingsRegistry.ts:197 Ctrl+Alt+ keybindings should not be used by default under Windows. Offender:  e  for  terminal.run
    e._assertNoCtrlAlt @ keybindingsRegistry.ts:197

    keybindingsRegistry.ts:197 Ctrl+Alt+ keybindings should not be used by default under Windows. Offender:  e  for  terminal.stop
    e._assertNoCtrlAlt @ keybindingsRegistry.ts:197

    keybindingsRegistry.ts:197 Ctrl+Alt+ keybindings should not be used by default under Windows. Offender:  e  for  terminal.open
    e._assertNoCtrlAlt @ keybindingsRegistry.ts:197

    keybindingsRegistry.ts:197 Ctrl+Alt+ keybindings should not be used by default under Windows. Offender:  e  for  terminal.toggle
    e._assertNoCtrlAlt @ keybindingsRegistry.ts:197

    extensionHost.ts:384 [Extension Host] Congratulations, your extension "terminal" is now active!

    extensionHost.ts:384 [Extension Host] vscode-icons is active!

**Comment by bpasero** [comment]
@DanAtkinson yeah good catch, this line seems to be where the trouble is coming from:

`[7220:0922/131701.662:ERROR:jump_list.cc(274)] Failed to append custom category 'Recent Workspaces' to Jump List due to system privacy settings.`

I am not sure however how to fix this, maybe someone else has an idea.

**Comment by DanAtkinson** [comment]
I've found out the issue now and have resolved it.

I did a search around jump lists and privacy and found [this article](https://winaero.com/blog/disable-jump-lists-windows-10/).

The steps to resolve are:
1. Open Settings.
2. Choose 'Personalisation'.
3. Choose 'Start'.
4. Ensure that 'Show recently opened items in Jump Lists on Start or the taskbar' is toggled to 'On' (as pictured below):
![image](https://user-images.githubusercontent.com/23440/30745149-fa321ea8-9f9c-11e7-86bf-6c2ce4d07c77.png)

For me, that toggle was switched to Off.

Once I'd closed and re-opened Code, the 'Recent Workspaces' populated itself.

Thanks for your help!
